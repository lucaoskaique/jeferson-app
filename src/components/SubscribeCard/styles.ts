import styled, { css } from 'styled-components'

export const Wrapper = styled.main`
  ${({ theme }) => css`
    background: ${theme.colors.primary};
    border-radius: ${theme.border.radius};
    padding: ${theme.spacings.xsmall};
    display: grid;
    height: 29.7rem;
  `}
`
export const Content = styled.div`
  ${({ theme }) => css`
    text-align: center;
    padding: ${theme.spacings.xsmall};
  `}
`

export const Title = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    font-weight: ${theme.font.bold};
    color: ${theme.colors.white};
  `}
`
export const Subtitle = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    font-weight: ${theme.font.light};
    color: ${theme.colors.white};
    margin-bottom: ${theme.spacings.xsmall};
  `}
`
