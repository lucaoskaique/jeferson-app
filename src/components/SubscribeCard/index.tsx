import Button from 'components/Button'

import * as S from './styles'

// export type SubscribeCardProps = {

// }

const SubscribeCard = () => (
  <S.Wrapper>
    <S.Content>
      <S.Title>Acopanhe nosso mandato!</S.Title>
      <S.Subtitle>Fique por dentro de not√≠cias e novidades</S.Subtitle>
      <div>
        <input></input>
        <input></input>
        <input></input>
        <input></input>
      </div>
      <Button size="small" background="white">
        enviar
      </Button>
    </S.Content>
  </S.Wrapper>
)

export default SubscribeCard
