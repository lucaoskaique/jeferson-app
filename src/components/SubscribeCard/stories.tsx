import { Story, Meta } from '@storybook/react'

import SubscribeCard from '.'

export default {
  title: 'SubscribeCard',
  component: SubscribeCard
} as Meta

export const Default: Story = () => <SubscribeCard />
