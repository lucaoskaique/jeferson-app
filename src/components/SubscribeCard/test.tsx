import { renderWithTheme } from 'utils/tests/helpers'

import SubscribeCard from '.'

describe('<SubscribeCard />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<SubscribeCard />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
