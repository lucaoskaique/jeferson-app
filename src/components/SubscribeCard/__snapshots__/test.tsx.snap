// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubscribeCard /> should render the heading 1`] = `
.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 0;
  cursor: pointer;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  border-radius: 0.4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 600;
  height: 3rem;
  font-size: 1.2rem;
  background: '#FAFAFA';
}

.c4:hover {
  background: linear-gradient(90deg,#72c042 -99.98%,#0092d3 193.73%);
}

.c0 {
  background: #EC6330;
  border-radius: 0.4rem;
  padding: 1.6rem;
  display: grid;
  height: 29.7rem;
}

.c1 {
  text-align: center;
  padding: 1.6rem;
}

.c2 {
  font-size: 1.6rem;
  font-weight: 600;
  color: #FAFAFA;
}

.c3 {
  font-size: 1.6rem;
  font-weight: 300;
  color: #FAFAFA;
  margin-bottom: 1.6rem;
}

<main
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      Acopanhe nosso mandato!
    </div>
    <div
      class="c3"
    >
      Fique por dentro de not√≠cias e novidades
    </div>
    <div>
      <input />
      <input />
      <input />
      <input />
    </div>
    <button
      class="c4"
    >
      <span>
        enviar
      </span>
    </button>
  </div>
</main>
`
